File: uikit/src/main/java/com/cometchat/pro/uikit/Sticker/StickerFragment.java
package com.cometchat.pro.uikit.Sticker;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.cometchat.pro.uikit.R;
import com.cometchat.pro.uikit.Sticker.listener.StickerClickListener;
import com.cometchat.pro.uikit.Sticker.model.Sticker;

import java.util.ArrayList;
import java.util.List;

import adapter.StickersAdapter;
import listeners.ClickListener;
import listeners.RecyclerTouchListener;

/**
 * Purpose - It is a Fragment which is used in StickerView to display stickers sent or received to a
 * particular receiver.
 *
 * Created On - 20th March 2020
 *
 * Modified On - 23rd March 2020
 *
 */
public class StickerFragment extends Fragment {

    private RecyclerView rvStickers;
    private StickersAdapter adapter;
    private String Id;
    private String TAG = "StickerView";
    private String type;
    private List<Sticker> stickers = new ArrayList<>();
    StickerClickListener stickerClickListener;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_stickers_view,container,false);
        rvStickers = view.findViewById(R.id.rvStickers);
        rvStickers.setLayoutManager(new GridLayoutManager(getContext(),4));
        Id = this.getArguments().getString("Id");
        type = this.getArguments().getString("type");
        List<Sticker> list = this.getArguments().getParcelableArrayList("stickerList");
        stickers = list;
        Log.e("onStickerView: ",stickers.get(0).getSetName());
        adapter = new StickersAdapter(getContext(),stickers);
        rvStickers.setAdapter(adapter);

        rvStickers.addOnItemTouchListener(new RecyclerTouchListener(getContext(), rvStickers, new ClickListener() {
            @Override
            public void onClick(View var1, int var2) {
                Sticker sticker = (Sticker)var1.getTag(R.string.sticker);
                stickerClickListener.onClickListener(sticker);
            }
        }));
        return view;
    }

    public void setStickerClickListener(StickerClickListener stickerClickListener) {
        this.stickerClickListener = stickerClickListener;
    }
}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/sticker/StickerFragment.kt
package com.cometchat.pro.uikit.sticker

import adapter.StickersAdapter
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.cometchat.pro.uikit.R
import com.cometchat.pro.uikit.sticker.listener.StickerClickListener
import com.cometchat.pro.uikit.sticker.model.Sticker
import listeners.ClickListener
import listeners.RecyclerTouchListener
import java.util.*

class StickerFragment : Fragment() {
    private var rvStickers: RecyclerView? = null
    private var adapter: StickersAdapter? = null
    private var Id: String? = null
    private val TAG = "StickerView"
    private var type: String? = null
    private var stickers: List<Sticker> = ArrayList<Sticker>()
    var stickerClickListener: StickerClickListener? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_stickers_view, container, false)
        rvStickers = view.findViewById(R.id.rvStickers)
        rvStickers?.layoutManager = GridLayoutManager(context, 4)
        Id = this.arguments!!.getString("Id")
        type = this.arguments!!.getString("type")
        val list: List<Sticker>? = this.arguments!!.getParcelableArrayList("stickerList")
        stickers = list!!
        Log.e("onStickerView: ", stickers[0].setName)
        adapter = StickersAdapter(context, stickers)
        rvStickers?.adapter = adapter

        rvStickers!!.addOnItemTouchListener(RecyclerTouchListener(context, rvStickers!!, object : ClickListener() {
            override fun onClick(var1: View, var2: Int) {
                val sticker = var1.getTag(R.string.sticker) as Sticker
                stickerClickListener!!.onClickListener(sticker)
            }
        }))
        return view
    }

    @JvmName("setStickerClickListener1")
    fun setStickerClickListener(stickerClickListener: StickerClickListener?) {
        this.stickerClickListener = stickerClickListener
    }
}